### PR TITLE
fix bugs of nodeMaxAvailableReplica

### PR DESCRIPTION
**Pull request by Garrybest**
Signed-off-by: Garrybest <garrybest@foxmail.com>

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/estimator/server/estimate.go [hunk 1]
```diff
@@ -72,8 +73,6 @@ func (es *AccurateSchedulerEstimatorServer) estimateReplicas(
 
 func (es *AccurateSchedulerEstimatorServer) nodeMaxAvailableReplica(node *framework.NodeInfo, rl corev1.ResourceList) int32 {
 	rest := node.Allocatable.Clone().SubResource(node.Requested)
-	if rest.AllowedPodNumber > int64(len(node.Pods)) {
-		rest.AllowedPodNumber -= int64(len(node.Pods))
-	}
+	rest.AllowedPodNumber = util.MaxInt64(rest.AllowedPodNumber-int64(len(node.Pods)), 0)
```
Comment (RainbowMango) [inline pkg/estimator/server/estimate.go:76]: Does the pod number on the node not included in `node.Requested`?

Comment (RainbowMango) [reply pkg/estimator/server/estimate.go:76]: Remind me what the bug is?

Comment (Garrybest) [reply pkg/estimator/server/estimate.go:76]: > Does the pod number on the node not included in node.Requested?

Pod number is included in `node.status`, but does not exist in `pod.spec`. So pod number should be handled specially.

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Comment by XiShanYongYe-Chang** [comment]
/release-note-edit ```karmada-estimator: fix bugs of nodeMaxAvailableReplica```

**Pull request merged**